UI/ChatUI/ChatUI/ChatUI/Entities/Message.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+
+namespace ChatUI.Entities;
+
+public record Message(string Text, string ContactName, DateTimeOffset Timestamp, bool IsMyMessage) : INotifyPropertyChanged
```

Comment:
As a good practice, a `record` should be immutable, do not allow any `set` (use only `init`) and do not implement `INotifyPropertyChanged`. For a MVVM example like you did (opposed to an MVUX), convert this to a `class`.